contrib/babelfishpg_tds/src/backend/tds/tdsresponse.c

```diff
@@ -517,29 +516,37 @@ is_numeric_cast(Oid func_oid)
 }
 
 /*
- * is_numeric_datatype - returns bool if given datatype is numeric or decimal.
+ * is_numeric_datatype - returns bool if given datatype is numeric, decimal, UDT on numeric or decimal.
  */
 static bool
 is_numeric_datatype(Oid typid)
 {
-	if (typid == NUMERICOID)
-	{
+	if (OidIsValid(typid) && getBaseType(typid) == NUMERICOID)
 		return true;
-	}
-	if (!OidIsValid(decimal_oid))
-	{
-		TypeName *typename = makeTypeNameFromNameList(list_make2(makeString("sys"), makeString("decimal")));
-		decimal_oid = LookupTypeNameOid(NULL, typename, false);
-	}
-	return decimal_oid == typid;
+
+	return false;
 }
 
-/* look for a typmod to return from a numeric expression */
+
+/* 
+ * look for a typmod to return from a numeric expression,
+ * Also for cases where we cannot compute the expression typmod return -1 and set found as false.
+ */
 int32
-resolve_numeric_typmod_from_exp(Plan *plan, Node *expr)
+resolve_numeric_typmod_from_exp(Plan *plan, Node *expr, bool *found)
 {
+	/*
+	 * set found value as true by default, if we are unable to 
+	 * find the expression typmod found will be set to false.
+	 */
+	if (found != NULL)
+		*found = true;
+
 	if (expr == NULL)
```

Comment:
When will expr be NULL?

Reply:
This is just a sanity check, that was already there.